Integrations/FortiSIEM/FortiSIEM.py

```diff
@@ -0,0 +1,690 @@
+    while response.text != "100" and max_wait_time > 0:
+        response = session.post(restAddress + '/reportProgress', headers=headers, data=data, verify=VERIFY_SSL)
+        max_wait_time = int(max_wait_time) - 1
+        time.sleep(1)
+
+    params = {
+        'start': 0,
+        'perPage': max_results,
+        'allData': extended_data,
+    }
+
+    response = session.post(restAddress + '/resultByReport', params=params, headers=headers, data=data,
+                            verify=VERIFY_SSL)
+
+    try:
+        res = response.json()
+        eventKeys = res["headerData"]["columnNames"]
+    except ValueError:
+        return_error("Got wrong response format when getting report results. "
+                     "Expected a json object but got " + response.text)
+        sys.exit(0)
+
+    # reformat results
+    eventData = []
+    md = ""
+    for key in res["lightValueObjects"]:
+        cur = {
+            'Event ID': key.get("naturalId", ""),
+            'Incident ID': incident_id,
+        }
+        for i in range(0, len(eventKeys)):
+            if len(key["data"]) == 0 or key["data"][0] == "No report results found.":
+                md = "No report results found."
+                break
+            else:
+                cur[eventKeys[i]] = key["data"][i]
+        if md != "":
+            # no results were found, not need to loop
+            break
+        cur["ExtendedData"] = {}
+        for extItem in key["extData"]:
+            if EXTENDED_KEYS.get(extItem["left"]) is not None:
+                cur[EXTENDED_KEYS.get(extItem["left"]).replace(' ', '')] = extItem["right"]
+            else:
+                cur["ExtendedData"][extItem["left"]] = extItem["right"]
+        eventData.append(cur)
+    md = tableToMarkdown(tableTitle, eventData, eventKeys) if md == "" else md
+
+    demisto.results({
+        'ContentsFormat': formats['json'],
+        'Type': entryTypes['note'],
+        'Contents': res,
+        'ReadableContentsFormat': formats['markdown'],
+        'HumanReadable': md,
+        'EntryContext': {'FortiSIEM.Events(val["Event ID"] && val["Event ID"] == obj["Event ID"])': eventData}
+    })
+
+
+def GetEventQuery(inXml):
+    Urlfirst = URL + "eventQuery"
+    headers = {'Content-Type': 'text/xml'}
+    resp = requests.request('POST', Urlfirst, headers=headers, data=inXml, verify=VERIFY_SSL, auth=AUTH)
+    queryId = resp.text
+    validateSuccessfulResponse(resp, "fetching event query")
+    if 'error code="255"' in queryId:
+        return_error("Got error code 255 getting event query. Make sure the query has valid syntax")
+        sys.exit(0)
+    return queryId
+
+
+def GetIncidentsByOrg(queryId):
+    # The request will poll until the server completes the query.
+    UrlSecond = URL + "progress/" + queryId
+    resp = requests.request('GET', UrlSecond, verify=VERIFY_SSL, auth=AUTH)
+
+    while resp.text != '100':
```

Comment:
100%
all query calls in fortisiem are done in 3 steps:
start the report (which is basically the query)
poll the report
get the results

it's quite fast and this is also how the UI works (at least 3-4 calls for every query)

I'm not familiar with the polling mechanism, but I think it will be redundant here since it's really a matter of 1-2 seconds.